src/rest/server_service/error.rs

```diff
@@ -0,0 +1,43 @@
+use native_tls::Error as TlsError;
+use std::error::Error as StdError;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::io::Error as IoError;
+
+#[derive(Debug)]
+pub enum Error {
+    Pkcs12LoadFailed(IoError),
+    Pkcs12Invalid(TlsError),
+    BindFailed(IoError),
+    ServerAlreadyStopped,
+    ServerStopTimeout,
+    ServerStopFailed,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let error_name = match *self {
+            Error::Pkcs12LoadFailed(_) => "Pkcs12LoadFailed",
+            Error::Pkcs12Invalid(_) => "Pkcs12Invalid",
+            Error::BindFailed(_) => "BindFailed",
+            Error::ServerAlreadyStopped => "ServerAlreadyStopped",
+            Error::ServerStopTimeout => "ServerStopTimeout ",
+            Error::ServerStopFailed => "ServerStopFailed ",
+        };
+        write!(f, "Server service error: {}", error_name)?;
```

Comment:
The choice of a prefix message should be left to a higher-level error handling code which knows it has this error type to represent. Just use a human-readable string describing the problem.

Reply:
Will fix in next PR